### PR TITLE
java: move java_dependency into a separate file and add Language::Java.java_home_env

### PR DESCRIPTION
**Pull request by xu-cheng**
Also update activemq to test against it

Per https://github.com/Homebrew/homebrew/pull/37192

@mikemcquaid I decide to create this PR first. After we sort out the Java part, I will then go to AmazonWebServicesFormula.

### CODE REVIEW

File: Library/Homebrew/language/java.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+module Language
+  module Java
+    def self.java_home_env(version=nil)
+      if version
+        { :JAVA_HOME => "$(/usr/libexec/java_home --version #{version})" }
+      else
+        { :JAVA_HOME => "$(/usr/libexec/java_home)" }
+      end
```
Comment (mikemcquaid) [inline Library/Homebrew/language/java.rb]: Could do `version_flag = " --version #{version}" if version`?

File: Library/Homebrew/requirements.rb [hunk 2]
```diff
@@ -122,46 +123,3 @@ class GitDependency < Requirement
   satisfy { !!which('git') }
 end
 
-class JavaDependency < Requirement
```
Comment (mikemcquaid) [inline Library/Homebrew/requirements.rb]: Has anything changed in this class?

Comment (xu-cheng) [inline Library/Homebrew/requirements.rb]: Nope.

File: Library/Homebrew/language/java.rb (hunk 1 above)
Comment (xu-cheng) [inline Library/Homebrew/language/java.rb]: Updated.

**Comment by mikemcquaid** [comment]
:+1:

**Comment by xu-cheng** [comment]
@BrewTestBot test this please

**Pull request closed without merging**